incubator/virtualcluster/experiment/pkg/scheduler/scheduler.go

```diff
@@ -157,10 +210,14 @@ func (s *Scheduler) Run(stopChan <-chan struct{}) {
 			return
 		}
 
-		klog.V(5).Infof("starting scheduler workers")
+		klog.Infof("starting scheduler virtualcluster workers")
 		for i := 0; i < s.virtualClusterWorkers; i++ {
 			go wait.Until(s.virtualClusterWorkerRun, 1*time.Second, stopChan)
 		}
+		klog.Infof("starting scheduler virtualcluster workers")
```

Comment:
split into two go routine. waitForCache+start worker for each goroutine.
and you are missing `defer s.superClusterQueue.ShutDown() ` here

Reply:
Good catch! Fixed.